slip-0019.md

```diff
@@ -5,24 +5,182 @@ Number:  SLIP-0019
 
-This is a section for a body. The title of the section should be changed
-and the section can be split into multiple sections and subsections.
+To mitigate such an attack, the hardware wallet needs to ascertain non-ownership of all inputs which are claimed to be external. In case of hierarchical deterministic wallets it is generally not feasible to ascertain this solely based on the scriptPubKey of the UTXO, because it would require searching through billions of BIP32 derivation paths. Furthermore, even though CoinJoin transactions currently work only with P2WPKH addresses, other applications may require more complicated inputs including multi-signature and Schnorr multi-signature in the future.
+
+A CoinJoin coordinator can also benefit from such a proof to verify that the CoinJoin participant is able and willing to sign the input. This verification helps to mitigate denial-of-service attacks as the attacker has to use a limited UTXO set that they control and in case of misbehavior this UTXO set gets banned.
+
+## Proof of ownership format
+
+A proof of ownership consists of a proof body and a signature. The proof body contains one or more ownership identifiers which allow a wallet to efficiently determine whether or not it is able to spend a UTXO having a given scriptPubKey. The proof signature affirms that the proof body can be trusted to have been generated by the true owner of the UTXO.
+
+```
+proofOfOwnership = proofBody || proofSignature
+```
+
+### Ownership identifier
+
+Let *k* be a secret *ownership identification key* derived from the wallet's master secret using the [SLIP-0021](https://github.com/satoshilabs/slips/blob/master/slip-0021.md) method for hierarchical derivation of symmetric keys as:
+
+```
+k = Key(m/"SLIP-0019"/"Ownership identification key")
+```
+
+The ownership identifier for a scriptPubKey is computed as:
+
+```
+id = HMAC-SHA256(key = k, msg = scriptPubKey)
+```
+
+In case of *m*-of-*n* multi-signature scriptPubKeys the proof of ownership SHOULD contain the ownership identifiers of all *n* co-owners of that scriptPubKey. See [Identifier inclusion](#identifier-inclusion) for further details.
+
+A wallet MUST NOT produce and reveal an ownership identifier for a scriptPubKey which it does not control. Such a fake ownership identifier can be used to mount a denial-of-service attack.
+
+### Proof body
+
+The *proofBody* is a concatenation of the following fields:
+
+* *version* (4 bytes): b"\x53\x4c\x00\x19".
```

Comment:
why these bytes?

Reply:
2f84a7a: Renamed it to versionMagic and added an explanation.